components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -10480,7 +10480,74 @@ public void addSubscriptionPolicy(SubscriptionPolicy policy) throws APIManagemen
                 handleException("Failed to add Subscription Policy: " + policy, e);
             }
         } finally {
-            APIMgtDBUtil.closeAllConnections(policyStatement, conn, null);
+            APIMgtDBUtil.closeAllConnections(policyStatement, conn, rs);
+        }
+    }
+
+    /**
+     * Get details of the subscription block condition by condition value and tenant domain
+     *
+     * @param conditionValue condition value of the block condition
+     * @param tenantDomain tenant domain of the block condition
+     * @return Block condition
+     * @throws APIManagementException
+     */
+    public BlockConditionsDTO getSubscriptionBlockCondition(String conditionValue, String tenantDomain)
+            throws APIManagementException {
+        Connection connection = null;
+        PreparedStatement selectPreparedStatement = null;
+        ResultSet resultSet = null;
+        BlockConditionsDTO blockCondition = null;
+        try {
+            String query = SQLConstants.ThrottleSQLConstants.GET_SUBSCRIPTION_BLOCK_CONDITION_BY_VALUE_AND_DOMAIN_SQL;
+            connection = APIMgtDBUtil.getConnection();
+            connection.setAutoCommit(true);
+            selectPreparedStatement = connection.prepareStatement(query);
+            selectPreparedStatement.setString(1, conditionValue);
+            selectPreparedStatement.setString(2, tenantDomain);
+            resultSet = selectPreparedStatement.executeQuery();
+            if (resultSet.next()) {
+                blockCondition = new BlockConditionsDTO();
+                blockCondition.setEnabled(resultSet.getBoolean("ENABLED"));
+                blockCondition.setConditionType(resultSet.getString("TYPE"));
+                blockCondition.setConditionValue(resultSet.getString("VALUE"));
+                blockCondition.setConditionId(resultSet.getInt("CONDITION_ID"));
+                blockCondition.setTenantDomain(resultSet.getString("DOMAIN"));
+                blockCondition.setUUID(resultSet.getString("UUID"));
+            }
+        } catch (SQLException e) {
+            if (connection != null) {
+                try {
+                    connection.rollback();
+                } catch (SQLException ex) {
+                    handleException("Failed to rollback getting Subscription Block condition with condition value "
+                            + conditionValue + " of tenant " + tenantDomain, ex);
+                }
+            }
+            handleException("Failed to get Subscription Block condition with condition value " + conditionValue
+                    + " of tenant " + tenantDomain, e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(selectPreparedStatement, connection, resultSet);
+        }
+        return blockCondition;
+    }
+
+    private void addGraphQLQueryAnalysisInfo(Connection conn, int maxDepth, int maxComplexity, int policyId)
```

Comment:
Is this related to JWT subscription block fix?

Reply:
nope ti think it came after i took a pull